src/rosdep2/rospkg_loader.py

```diff
@@ -117,6 +117,33 @@ def get_loadable_resources(self):
             self._loadable_resource_cache = list(self._rospack.list())
         return self._loadable_resource_cache
 
+    def dep_version_checking(self, deps):
+        names_with_version = []
+        for i in (0, len(deps) - 1):
+            version_values = []
+            try:
+                version_values.append(getattr(deps[i], "version_eq"))
+                version_values.append(getattr(deps[i], "version_gt"))
+                version_values.append(getattr(deps[i], "version_gte"))
+                version_values.append(getattr(deps[i], "version_lt"))
+                version_values.append(getattr(deps[i], "version_lte"))
+            except:
+                print("ERROR: Dependencies version checking has failed,")
+                print("if no version tag has been provided this message can be ignored.")
```

Comment:
Actually there is no specific reason for catching an error here, it was just in case the attribute names of object were changed, I think the best option is to remove this "try except", because if attribute name is changed this code should be maintain also. Am I right ?